test/WebJobs.Script.Tests/NodeEndToEndTests.cs

```diff
@@ -703,13 +703,26 @@ public void ExcludedFunction_NotAddedToHost()
                     { "input", input.ToString() }
                 });
 
-            await TestHelpers.Await(() =>
+            await Task.WhenAny(t, Task.Delay(5000));
+            if (!t.IsCompleted) throw t.Exception;
+        }
+
+        [Fact]
+        public async Task PromiseResolve()
+        {
+            JObject input = new JObject
             {
-                return t.IsCompleted;
-            }, timeout: 5000, pollingInterval: 1000);
+                { "scenario", "promiseResolve" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
 
-            // Await the task to force any exception to be thrown
-            await t;
+            await Task.WhenAny(t, Task.Delay(5000));
+            if (!t.IsCompleted) throw t.Exception;
```

Comment:
By the way, for completeness, the typical pattern (which this *could* be updated to use) is to just make sure that the `Task` that triggered the transition (the task that completed first) is the expected `Task`. Example
``` csharp
Task result = await Task.WhenAny(t, Task.Delay(5000));

Assert.Same(t, result);
// or, to have a descriptive message
Assert.True(ReferenceEquals(t, result), "Some timeout message...");

Then you can still throw if your task is faulted.